### ISSUE TITLE
Bug in callbacks

### ISSUE THREAD

**Issue by certik** [lfortran/lfortran#1163]
```fortran
module XX

contains

subroutine find_fit(x, expr)
    real :: x
    interface
        function expr(x) result(y)
            real, intent(in) :: x
            real :: y
        end function
    end interface
end subroutine

end module

program main
    use XX, only: find_fit
    real :: x
    call find_fit(x, f)
    contains
    function f(x) result(y)
        real, intent(in) :: x
        real :: y
    end function
end program
```
This works with gfortran but fails in lfortran:
```console
lfortran/lfortran(main) $ gfortran a.f90
lfortran/lfortran(main) $ lfortran a.f90
ASR verify pass error: ASR verify: Var::m_v `expr` cannot point outside of its symbol table
Internal Compiler Error: Unhandled exception
Traceback (most recent call last):
  File "/Users/ondrej/repos/lfortran/lfortran/src/bin/lfortran.cpp", line 1932
    err = compile_to_object_file(arg_file, tmp_o, false,
  File "/Users/ondrej/repos/lfortran/lfortran/src/bin/lfortran.cpp", line 862
    res = fe.get_llvm3(*asr, lpm, diagnostics, infile);
  File "/Users/ondrej/repos/lfortran/lfortran/src/lfortran/fortran_evaluator.cpp", line 334
    compiler_options, run_fn, infile);
  File "/Users/ondrej/repos/lfortran/lfortran/src/libasr/codegen/asr_to_llvm.cpp", line 6700
    pass_manager.apply_passes(al, &asr, pass_options, diagnostics);
  File "/Users/ondrej/repos/lfortran/lfortran/src/libasr/pass/pass_manager.h", line 201
    _apply_passes(al, asr, _passes, pass_options, diagnostics);
  File "/Users/ondrej/repos/lfortran/lfortran/src/libasr/pass/pass_manager.h", line 102
    throw LFortran::LCompilersException("Verify failed");
LCompilersException: Verify failed
```

**Comment by Pranavchiku** [comment]
Okay so this is because of pass `pass_unused_functions` which removes unused functions from ASR, here in this `expr` is such function and hence it throws the above log.

**Comment by Pranavchiku** [comment]
I guess we can resolve this by checking if Unused Function `expr` is in dependency list of Function `find_fit`, then don't remove it from ASR.

**Comment by certik** [comment]
I think we have to add an exception for function (interfaces) that are used in the function argument (parameter) list.

**Comment by Pranavchiku** [comment]
Fixed by #1178.
